src/pages/en/guides/troubleshooting.md

```diff
@@ -0,0 +1,181 @@
+
+Then check your import statement:
+
+- Is your import linking to the wrong place? (Check your import path.)
+
+- Does your import have the same name as the imported component? (Check your component name and that it [follows the `.astro` syntax](/en/comparing-astro-vs-other-tools/#astro-vs-jsx).)
+
+- Have you included the extension in the import? (Check that your imported file contains an extension. e.g. `.astro`, `.md`, `.jsx`, `.vue`)
+
+### My component is not interactive
+
+If your component is rendering (see above) but is not responding to user interaction, then you may be missing a [`client:*` directive](/en/reference/directives-reference/#client-directives) to hydrate your component.
+
+By default, a [UI Framework component is not hydrated in the client](/en/core-concepts/framework-components/#hydrating-interactive-components). If no `client:*` directive is provided, its HTML is rendered onto the page without JavaScript.
+
+Note: [Astro components](/en/core-concepts/astro-components/) are HTML-only templating components with no client-side runtime. But, you can use a `<script>` tag in your Astro component template to send JavaScript to the browser that executes in the global scope.
+
+### Cannot find package 'X'
+
+See [Astro's integration guide](/en/guides/integrations-guide/) for instructions on adding framework renderers, CSS tools and other packages to Astro.
+
+You may need to install peer dependencies for some integrations. If you see a “missing peer dependencies” warning, you can follow the instructions for  [handling dependencies](/en/guides/integrations-guide/#handling-integration-dependencies).
+
+### `Astro.glob()`'s "No matches found"
+
+When using `Astro.glob()` to import files, be sure to use the correct glob syntax that will match all the files you need.
+
+For example, use `src/components/**/*.js` to import both of the following files:
+- `src/components/MyComponent.js`
+- `src/components/includes/MyOtherComponent.js`
+
+> Note that dynamic variables and string interpolation in `Astro.glob()` is not supported.
+
+This is not a bug in Astro. It is due to a limitation of [Vite's `import.meta.glob()` function](https://vitejs.dev/guide/features.html#glob-import) which only supports static string literals.
+
+A common workaround is to instead import a larger set of files that includes all the files you need using `Astro.glob()`, then filter them:
+
+```astro
+---
+// src/components/featured.astro
+const { postSlug } = Astro.props
+const pathToMyFeaturedPost = `src/pages/blog/${postSlug}.md`
+
+const posts = await Astro.glob('../pages/blog/*.md');
+const myFeaturedPost = posts.find(post => post.file.indexOf(pathToMyFeaturedPost) !== -1);
+---
+
+<p>
+    Take a look at my favorite post, <a href={myFeaturedPost.url}>{myFeaturedPost.frontmatter.title}</a>!
+</p>
+```
```

Comment:
Should the blockquote maybe come at the end of this section rather than in the middle?

I’ve also updated the `posts.find` line for a slightly more readable alternative.

````suggestion
This is not a bug in Astro. It is due to a limitation of [Vite's `import.meta.glob()` function](https://vitejs.dev/guide/features.html#glob-import) which only supports static string literals.

A common workaround is to instead import a larger set of files that includes all the files you need using `Astro.glob()`, then filter them:

```astro
---
// src/components/featured.astro
const { postSlug } = Astro.props
const pathToMyFeaturedPost = `src/pages/blog/${postSlug}.md`

const posts = await Astro.glob('../pages/blog/*.md');
const myFeaturedPost = posts.find(post => post.file.includes(pathToMyFeaturedPost));
---

<p>
    Take a look at my favorite post, <a href={myFeaturedPost.url}>{myFeaturedPost.frontmatter.title}</a>!
</p>
```

> Note that dynamic variables and string interpolation in `Astro.glob()` is not supported.
````

Reply:
I think "This is not a bug"  and the workaround example refers to that call out, so I'm not sure that it makes sense to move that below?

Is the callout distracting, or does it seem out of place?

Reply:
OH! If the blockquote and the following stuff is _connected_ I think they should either both be in the blockquote or the blockquote shouldn’t be there. I see the blockquote as separating out everything inside from what’s outside.

Reply:
Yeah, look at the changes I commited to the branch and see if that reads better?

Reply:
And… I just saw you already fixed this 💜